api/v1alpha1/flinkcluster_validate.go

```diff
@@ -55,14 +55,45 @@ func (v *Validator) ValidateCreate(cluster *FlinkCluster) error {
 
 // ValidateUpdate validates update request.
 func (v *Validator) ValidateUpdate(old *FlinkCluster, new *FlinkCluster) error {
+	var cancelRequested, err = v.checkCancelRequested(old, new)
+	if err != nil {
+		return err
+	}
+	if cancelRequested {
+		return nil
+	}
+
 	if !reflect.DeepEqual(new.Spec, old.Spec) {
-		return fmt.Errorf(
-			"updating FlinkCluster spec is not allowed," +
-				" please delete the resouce and recreate")
+		return fmt.Errorf("the cluster properties are not updatable")
 	}
+
 	return nil
 }
 
+func (v *Validator) checkCancelRequested(
+	old *FlinkCluster, new *FlinkCluster) (bool, error) {
+	if old.Spec.Job == nil || new.Spec.Job == nil {
+		return false, nil
+	}
+	var restartJob = (old.Spec.Job.CancelRequested != nil && *old.Spec.Job.CancelRequested) &&
+		(new.Spec.Job.CancelRequested == nil || !*new.Spec.Job.CancelRequested)
+	if restartJob {
+		return false, fmt.Errorf(
+			"updating cancelRequested from true to false is not allowed")
```

Comment:
Why cant this be just restart is not allowed after cancle?

Reply:
We'll support restart after finishing #84.